codelists/migrations/0034_auto_20210429_1029.py

```diff
@@ -0,0 +1,24 @@
+# Generated by Django 3.1.8 on 2021-04-29 10:29
+
+from django.db import migrations
+
+
+def create_handles(apps, schema_editor):
+    Codelist = apps.get_model("codelists", "Codelist")
+
+    for cl in Codelist.objects.all():
+        cl.handles.create(
+            name=cl.name,
+            slug=cl.slug,
+            organisation=cl.organisation,
+            user=cl.user,
+            is_current=True,
+        )
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("codelists", "0033_auto_20210429_1029"),
+    ]
+
+    operations = [migrations.RunPython(create_handles)]
```

Comment:
Can you add a reverse step to this (even if it's a no-op)?  Makes the migration a little easier to use 🙂

Reply:
Good idea.

Reply:
I think I'm going to leave this.  The [next migration (0035)](https://github.com/opensafely-core/opencodelists/pull/683/files#diff-bf77995fe00a8f8fe70a14fbb5c14e0194bf2e4b3ada400006ed6fc2ee16d91d) removes a bunch of fields from `Codelist`, some of which are required.  As such, 0035 is not reversible.  So to make a completely reversible chain of migrations, I'd have to have something like:

migration | forwards | backwards
-|-|-
0033 | add Handle model | remove Handle model (automatic)
0034 | make fields nullable on Codelist | make fields required on Codelist (automatic)
0035 | create Handle instance for each Codelist | populate Codelist fields from Handle
0036 | delete fields from Codelist | create fields on Codelist (automatic)

This is all doable, but given that we're not doing development here in parallel, I don't think it's worth it.